proposed/extended-coding-style-guide.md

```diff
@@ -871,29 +871,45 @@ try {
 
 ## 6. Operators
 
-All binary and ternary operators MUST be preceded and followed by at least
-one space; multiple spaces MAY be used for readability purpose. This includes all [arithmetic][],
-[comparison][], [assignment][], [bitwise][], [logical][] (excluding `!` which is unary),
-[string concatenation][], [type][] operators, trait operators (`insteadof` and `as`),
-and the single pipe operator (e.g. `ExceptionType1 | ExceptionType2 $e`).
+Style rules for operators are grouped by arity (the number of operands they take).
 
-There MUST NOT be any whitespace between the increment/decrement operators and the variable 
-being incremented/decremented.
+When space is permitted around an operator, multiple spaces MAY be
+used for readability purposes.
 
-Other operators are left undefined.
+Any operators not described here are left undefined.
 
-For example:
+### 6.1. Unary operators
+
+The increment/decrement operators MUST NOT have any space between
+the operator and operand.
 
+Type casting operators MUST NOT have any space within the parentheses:
 ~~~php
-<?php
+$intValue = (int) $input;
+~~~
 
+### 6.2. Binary operators
+
+All binary [arithmetic][], [comparison][], [assignment][], [bitwise][],
+[logical][], [string][], and [type][] operators MUST be preceded and
+followed by at least one space:
+~~~php
 if ($a === $b) {
     $foo = $bar ?? $a ?? $b;
 } elseif ($a > $b) {
-    $variable = $foo ? 'foo' : 'bar';
+    $foo = $a + $b * $c;
 }
 ~~~
 
+### 6.3. Ternary operators
+
+The conditional operator, also known simply as the ternary operator, MUST be
+preceded and followed by at least one space around both the `?`
+and `:` characters:
```

Comment:
What about the compressed (`?:`) form?

Reply:
Good catch! At that point, there's 2 operands. I think other languages call it the Elvis operator. PHP has it classified under comparison operators, so in a sense it's already being addressed by requiring space around binary comparison operators. But that is probably not explicit/intuitive enough to leave as is, as PHP also lets you put spaces/comments between the 2 characters `$foo ? /*this operand is optional*/ : $bar` while still behaving as a binary operator. So do we just add to the standard that when the middle operand is omitted, the operator MUST always be written as `?:` and MUST be preceded and followed by at least one space?

Reply:
:+1: for me, but we should check if it should be counted as a clarification or as a new rule. @dragonmantank WDYT?

Reply:
I removed the one part requiring that it MUST be written as `?:` - that could definitely be seen as a new rule that the original document didn't specify, but I left the clarification that when the middle operand is left out, format it like other binary comparison operators. So the current version should contain no new rules, just clarifications and reformatting.